test/browser/imagetag.test.jsx

```diff
@@ -0,0 +1,29 @@
+var should = require('should');
+var stubRouterContext = require('./stub-router-context.jsx');
+
+var ImageTag = require('../../components/imagetag.jsx');
+
+describe('ImageTag', function () {
+  var
+      imagetag;
+
+  imagetag = stubRouterContext.render(ImageTag, {
+    src1x: 'foo1x.jpg',
+    src2x: 'foo2x.jpg'
+  });
+
+
+  it('should set pixel density to 1 when it is less than 1.5', function () {
+    var window = {
+      devicePixelRatio: 0.9
+    };
+    imagetag.state.pixelDensity.should.be['1'];
```

Comment:
Oh, I think the reason this test appears to be passing is because the end of the line should be a function call, not an array reference, i.e.:

```
imagetag.state.pixelDensity.should.be(1);
```

Otherwise, as an array reference it's just returning `undefined` with no side effects. I suspect that once you change it and line 27 to function calls, at least one of them should fail, because I don't think that your component has any way of accessing the local `window` object you're creating at lines 17 and 24.

If my prediction is correct, then one way to proceed is to use [Mocha's hooks](http://mochajs.org/#hooks) to save and restore the actual value of `window.devicePixelRatio` before and after test runs, and set `window.devicePixelRatio` within the tests.

If that doesn't work, e.g. because a host browser doesn't like you overwriting its stuff, then an alternative might be to pass a fake window object in as an additional prop, but use [default props](http://facebook.github.io/react/docs/reusable-components.html#default-prop-values) to have the global window be its default value, so that production code doesn't need to explicitly pass it in.